src/main/java/pl/cleankod/exchange/provider/NbpLFUCache.java

```diff
@@ -0,0 +1,49 @@
+package pl.cleankod.exchange.provider;
+
+import org.springframework.stereotype.Component;
+import pl.cleankod.exchange.core.gateway.Cache;
+import pl.cleankod.exchange.provider.model.CacheEntry;
+import pl.cleankod.exchange.provider.model.RateWrapper;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@Component
+public class NbpLFUCache implements Cache {
+
+    private Map<String, CacheEntry> cacheMap = new HashMap<>();
```

Comment:
Is the `HashMap` threadsafe?

What happens when multiple threads try to write the same entry?